sdk/core/src/main/java/com/klaviyo/coresdk/Klaviyo.kt

```diff
@@ -1,73 +1,146 @@
+ * Receives configuration, profile data, and analytics requests
+ * to be processed and sent to the Klaviyo backend
  */
 object Klaviyo {
 
     /**
-     * Assigns a user email address to the user info
-     * This email address is stored in memory to define user details during a session and will be
-     * used to autocomplete analytics requests that are missing the identifier in their property maps
+     * Configure Klaviyo SDK with your account's public API Key and application context.
+     * This must be called to before using any other SDK functionality
      *
-     * Whenever the active user in an app changes (for example via a fresh login) this should be
-     * called to set the new user information in memory.
-     *
-     * @param email Email address for an active user
+     * @param apiKey - Your Klaviyo account's public API Key
+     * @param applicationContext
+     * @return
      */
-    fun setUserEmail(email: String) {
-        UserInfo.email = email
+    fun initialize(apiKey: String, applicationContext: Context) {
+        Registry.config = Registry.configBuilder
+            .apiKey(apiKey)
+            .applicationContext(applicationContext)
+            .build()
     }
 
     /**
-     * Creates a track request for the Klaviyo APIs and begins processing.
-     * Track requests track the triggering of a specified [KlaviyoEvent] in Klaviyo
+     * Assigns an email address to the current internally tracked profile
+     *
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
+     *
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
      *
-     * @param customerProperties A map of customer property information.
-     * Defines the customer that triggered this event
-     * @param properties A map of event property information.
-     * Additional properties associated to the event that are not for identifying the customer
+     * @param email Email address for active user
+     * @return
      */
-    fun track(
-        event: KlaviyoEvent,
-        customerProperties: KlaviyoCustomerProperties,
-        properties: KlaviyoEventProperties? = null
-    ) {
-        val request = TrackRequest(KlaviyoConfig.apiKey, event, customerProperties, properties)
-        processRequest(request)
-    }
+    fun setEmail(email: String): Klaviyo = this.setProfile(Profile().setEmail(email))
+
+    /**
+     * Assigns a phone number to the current internally tracked profile
+     *
+     * The SDK keeps track of current profile details to
+     * build analytics requests with profile identifiers
+     *
+     * This should be called whenever the active user in your app changes
+     * (e.g. after a fresh login)
+     *
+     * @param phoneNumber Phone number for active user
+     */
+    fun setPhoneNumber(phoneNumber: String): Klaviyo =
+        this.setProfile(Profile().setPhoneNumber(phoneNumber))
```

Comment:
nit, but I still find this a little hard to read. When I set set profile taking in a fresh `Profile` object which just has a phone number set, I imagine that I'm overwriting all profile data in storage. I know thats not what the function does, but when I read this as code it just fails to click in my mind until I dig into the `UserInfo` class and see what the logic actually does and I worry a little that it would have the same effect on some users of our SDK.

Reply:
I don't think it's a critical or blocking issue of any sort cause its something that could be pretty easily changed without affecting the contract between our SDK and the developer's app. It's all internal details.

But ya I just think maybe something like this might be more immediately consumable.
```kotlin
fun setPhoneNumber(phoneNumber: String): Klaviyo =
    UserInfo.phoneNumber = phoneNumber
    identify()
```

Where `identify` is just a call to enqueue the identify request after building the current profile from whats in `UserInfo` exactly as you currently do.

Reply:
Honestly it really doesn't affect much but thought I'd put down my rambled thoughts on it

Reply:
In a related vein I actually worry a bit that `setProfile` as a public method is confusing, because like you said it sounds like it would overwrite all profile data, not just set the values you're passing in.

Reply:
Thanks for this feedback, I agree it wasn't very readable. Check out this PR https://github.com/klaviyo/klaviyo-android-sdk/pull/35 I think it makes it clearer.